src/examples/obs_multi.cpp

```diff
@@ -40,53 +37,48 @@ struct Params {
     };
 };
 
-template <typename Params, typename Model>
-class UCB_multi {
-public:
-    UCB_multi(const Model& model, int iteration = 0) : _model(model) {}
-
-    size_t dim_in() const { return _model.dim_in(); }
-
-    size_t dim_out() const { return _model.dim_out(); }
-
-    double operator()(const Eigen::VectorXd& v) const
-    {
-        // double mu, sigma;
-        // std::tie(mu, sigma) = _model.query(v);
-        // return (mu + Params::ucb::alpha() * sqrt(sigma));
-
-        return (sqrt(_model.sigma(v)));
-    }
-
-protected:
-    const Model& _model;
-};
-
-struct fit_eval {
+struct StateEval {
     static constexpr size_t dim_in = 2;
     static constexpr size_t dim_out = 2;
 
     Eigen::VectorXd operator()(const Eigen::VectorXd& x) const
     {
         Eigen::VectorXd res(2);
-        res(0) = 0;
-        for (int i = 0; i < x.size(); i++)
-            res(0) += 1 - (x[i] - 0.3) * (x[i] - 0.3) + sin(10 * x[i]) * 0.2;
-        res(1) = res(0);
+        res(0) = 3 * x(0) + 5;
+        res(1) = -5 * x(1) + 2;
         return res;
     }
 };
 
+struct Average {
+    typedef double result_type;
+    double operator()(const Eigen::VectorXd& x) const
+    {
+        return (x(0) + x(1)) / 2;
```

Comment:
Minor detail but you should call it "Average2" or use x.sum()/x.length() (probably better)